src/rabbit_direct.erl

```diff
@@ -101,6 +105,45 @@ connect(Creds, VHost, Protocol, Pid, Infos) ->
         false -> {error, broker_not_found_on_node}
     end.
 
+extract_extra_auth_props(Creds, VHost, Pid, Infos) ->
+    case extract_protocol(Infos) of
+        undefined ->
+            [];
+        Protocol ->
+            maybe_call_connection_info_module(Protocol, Creds, VHost, Pid, Infos)
+    end.
+
+extract_protocol(Infos) ->
+    case proplists:get_value(protocol, Infos, undefined) of
```

Comment:
We should test that all protocols we currently support either don't have this entry or have a connection info module.